src/sensor_service.cpp

```diff
@@ -523,15 +528,13 @@ void SensorServiceClass::setConnectable(void)
 {
   tBleStatus ret;
 
-  const char local_name[] = {AD_TYPE_COMPLETE_LOCAL_NAME,'B','l','u','e','N','R','G'};
-
   if(set_connectable){
     /* disable scan response */
     hci_le_set_scan_resp_data(0,NULL);
     PRINTF("General Discoverable Mode.\n");
 
     ret = aci_gap_set_discoverable(ADV_IND, 0, 0, PUBLIC_ADDR, NO_WHITE_LIST_USE,
-                                   sizeof(local_name), local_name, 0, NULL, 0, 0);
+                                   strlen(dev_name), dev_name, 0, NULL, 0, 0);

```

Comment:
I think there is an issue here and/or maybe in the `begin()`
`dev_name` is not a NULL terminated string so the `strlen()` will return a wrong value, right?
Moreover, previously the size of the `local_name` was used (8) while in the begin() the `dev_name[0]` is not considered ? I know function are not the same, but I think this is something to verify because looking those function in aci gatt/gap, it seems handled to the same way.

Reply:
You are definitely right about strlen, I substituted it with AD + real because it seemed to me to be more correct.
About the difference between the two functions I agree with your perplexities, but I am not experienced enough in the bluetooth stack to give you a certain feedback. I followed the function structure as it was written before but I can't be sure it's the correct/better approach.

Reply:
Thanks @NoahRosa.
@cparata any thought on this?

Reply:
For me it is fine

Reply:
ok, you can merge and release.